include/TPP/Dialect/Tpp/TppOps.td

```diff
@@ -138,6 +138,53 @@ def Tpp_MatmulOp : Tpp_Op<"matmul"> {
     ```
   }];
 
+  let arguments = (ins TppOperand:$matrixA, TppOperand:$matrixB, 
+                       TppOperand:$matrixC);
+
+  let assemblyFormat = [{
+      `ins` `(` $matrixA `:` type($matrixA) `,` $matrixB `:` type($matrixB) `)`
+      `out` `(` $matrixC `:` type($matrixC) `)` attr-dict
+  }];
+
+  let extraClassDeclaration = [{
+    MemRefType getMatrixCType() { 
+      return getMatrixC().getType().cast<MemRefType>(); 
+    }
+    MemRefType getMatrixAType() { 
+      return getMatrixA().getType().cast<MemRefType>(); 
+    }
+    MemRefType getMatrixBType() { 
+      return getMatrixB().getType().cast<MemRefType>(); 
+    }
+  }];
+
+  let builders = [
+    OpBuilder<(ins "ValueRange":$inputs, "Value":$output)>
+  ];
+
+  let hasVerifier = 1;
+}
+
+//===----------------------------------------------------------------------===//
+// VNNI_MatmulOp
+//===----------------------------------------------------------------------===//
+
+def Tpp_VNNI_MatmulOp : Tpp_Op<"vnni_matmul"> {
+  let summary = "Performs matrix multiplication of two inputs, with the first operand in VNNI format.";
+
+  let description = [{
+    The `tpp.vnni_matmul` mirrors `vnni.matmul`.
```

Comment:
what is `vnni.matmul`? We have only `tpp.vnni_matmul`, right?

Reply:
I am adding vnni dialect, added the comment in anticipation of the op being added. I'll revert this for now.